src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinderProvider.cs

```diff
@@ -20,9 +19,7 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (context.Metadata.IsComplexType &&
-                !context.Metadata.IsCollectionType &&
-                HasDefaultConstructor(context.Metadata.ModelType.GetTypeInfo()))
+            if (context.Metadata.IsComplexType && !context.Metadata.IsCollectionType)
```

Comment:
After some offline thinking and talking through the scenarios, the current approach restores the ASP.NET Core 1.0 behaviour w.r.t. this provider and now throws an intelligible error where it should (in the binder). The provider you've got here is **Just Fine:tm:**.